lib/game_list_view/game_list_view.dart

```diff
@@ -1,203 +1,471 @@
+    required int count,
+    required VoidCallback onPressed,
+    String label = 'Filter hinzufügen',
+  }) {
+    return Stack(
+      clipBehavior: Clip.none,
+      children: [
+        ElevatedButton.icon(
+          onPressed: onPressed,
+          icon: const Icon(Icons.tune),
+          label: Text(label),
+        ),
+        if (count > 0)
+          Positioned(
+            right: -6,
+            top: -6,
+            child: Container(
+              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
+              decoration: BoxDecoration(
+                color: Colors.redAccent,
+                borderRadius: BorderRadius.circular(999),
+              ),
+              child: Text(
+                '$count',
+                style: const TextStyle(
+                  color: Colors.white,
+                  fontSize: 12,
+                  fontWeight: FontWeight.bold,
+                ),
+              ),
+            ),
+          ),
+      ],
+    );
+  }
+
+  Widget _buildDoubleFieldRow(
+    String leftLabel,
+    TextEditingController leftTextFieldController,
+    String rightLabel,
+    TextEditingController rightTextFieldController,
+    GameListController filterController,
+  ) {
+    return Row(
+      children: [
+        _buildNumberFilterField(
+          leftTextFieldController,
+          leftLabel,
+          filterController,
+        ),
+        const SizedBox(width: 10),
+        _buildNumberFilterField(
+          rightTextFieldController,
+          rightLabel,
+          filterController,
+        ),
+      ],
+    );
+  }
+
+  Expanded _buildNumberFilterField(
+    TextEditingController textFieldController,
+    String label,
+    GameListController filterController,
+  ) {
+    return Expanded(
+      child: TextField(
+        controller: textFieldController,
+        keyboardType: TextInputType.number,
+        decoration: InputDecoration(
+          labelText: label,
+          suffixIcon: IconButton(
+            icon: const Icon(Icons.clear),
+            onPressed: () => filterController.clearField(textFieldController),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildResultList(BuildContext context) {
+    final controller = Provider.of<GameListController>(context, listen: false);
+    return Expanded(
+      child: NotificationListener<ScrollNotification>(
+        onNotification: (notification) {
+          if (notification.metrics.extentAfter < 200) {
+            controller.loadMore();
+          }
+          return false;
+        },
+        child: Selector<GameListController, List<Game>>(
+          selector: (_, c) => c.visibleGames,
+          builder: (context, games, _) {
+            return ListView.builder(
+              itemCount: games.length,
+              itemBuilder: (context, index) {
+                final game = games[index];
+                return GameCard(
+                  game,
+                  key: ValueKey(game.identifier),
```

Comment:
Das hier ist von den Indentation Levels grenzwertig. Man könnte hier den Selector in eine eigene Funktion auslagern, dann wärs aufgeräumter.